Newsletters: add 160 (2021-08-04)

harding opened the pull request:
- [ ] Update releases/RCs on Tuesday @harding 
- [ ] Add entries to topic pages @harding 
- [ ] Bitcoin Core 22.0 RC2 description @jnewbery
- [ ] `Bitcoin Core #22006` first tracepoints and documentation on User-Space, Statically Defined Tracing @glozow 
- [ ] `Eclair #1893` Different default relay fees for announced and unannounced channels @bitschmidty 
- [ ] `Rust-Lightning #967` valentinewallace/2021-06-keysend @dongcarl

jnewbery commented on _includes/specials/taproot/en/06-multisignatures.md:
```diff
@@ -0,0 +1,147 @@
+a keypath spend, it is indistinguishable from single-sig spends.  If all
+single-sig and multisigs in the block above were switched to P2TR
+keypath spends, only a few exotic spends would be distinguishable by
+their scripts (and even some of those might be able to sometimes use
+keypath spends).
```
The 'taproot assumption' is that _all_ outputs can be spent using a multisignature keypath spend:

> there exists some monotone function on plain public keys and nothing else that is sufficient to authorize a transaction

(https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2018-February/015700.html)

Can we replace "even some of those might be able to use keypath spends" to "and even those could use keypath spends in the best case" or similar.

jnewbery commented on _includes/specials/taproot/en/06-multisignatures.md:
```diff
@@ -0,0 +1,147 @@
+  but which requires three rounds of communication during the signing
+  process.
+
+- **MuSig2**, also simple to implement and which eliminates one round of
+  communication and allows another round to be combined with key
+  exchange so that it's possible to use a somewhat similar signing
+  process to what we use today with script-based multisig.  This does
```
Long sentence could be split up:

```suggestion
- **MuSig2** is also simple to implement. It eliminates one round of
  communication and allows another round to be combined with key
  exchange. That means it's possible to use a somewhat similar signing
  process to what we use today with script-based multisig.  This does
```

jnewbery commented on _includes/specials/taproot/en/06-multisignatures.md:
```diff
@@ -0,0 +1,147 @@
+  hardware can't be tricked into unknowingly repeating part of the
+  signing session.
+
+- **MuSig-DN** (Deterministic Nonce), significantly more complex to
+  implement and its communication between participants can't be combined
+  with key exchange.  However, it's not vulnerable to the repeated
```
This sentence could also be split up.

jnewbery commented on _includes/specials/taproot/en/06-multisignatures.md:
```diff
@@ -0,0 +1,147 @@
+  with key exchange.  However, it's not vulnerable to the repeated
+  session attack.
+
+All signers have to agree on the protocol to use, so there will likely
+be a network effect where almost all implementations use the same
+protocol.  The authors of the MuSig proposals [suggest][nick ruffing
```
Maybe, although I think we're a very long way off being able to create generalised multisignatures across an arbitrary set of wallet implementations. I would have thought that most people using multisignature initially would just use a single wallet for all of the signers, which would mean that they could choose any multisignature scheme they want.

harding replied:
> I think we're a very long way off being able to create generalised multisignatures across an arbitrary set of wallet implementations.

Interesting!  I thought the opposite.  All the work projects like HWI have put into creating compatibility shims, plus the need for only small upgrades to PSBT, seems to me to imply that all that's left to do is for actual signing hardware/software to implement MuSigWhatever.  Surely that's two weeks work at most 😀

> I would have thought that most people using multisignature initially would just use a single wallet for all of the signers, which would mean that they could choose any multisignature scheme they want.

That probably is true.  But network effects travel back through time through the anticipation of those effects.  I think there are few wallet developers who want to spend their time implementing a different protocol than the protocol they expect to become widely used in the long term.

Anyway, I'll see if I can tone down the statement in the text.

jnewbery replied:
New wording looks good to me. Thanks!

jnewbery commented on _includes/specials/taproot/en/06-multisignatures.md:
```diff
@@ -0,0 +1,147 @@
+most likely to sign.  For example, in a 2-of-3 situation, it might be
+known that normally Alice and Bob will co-sign, while Carol only signs
+if one of the others is unavailable.  For this set of circumstances, the
+primary keys can use a multisignature for the taproot key-path spend (e.g.
```
```suggestion
primary keys can use a multisignature for the taproot keypath spend (e.g.
```

jnewbery commented on _includes/specials/taproot/en/06-multisignatures.md:
```diff
@@ -0,0 +1,147 @@
+
+In the normal case, the above has exactly as much efficiency and privacy
+as a single-sig or multisignature transaction.  In the abnormal case,
+spending still works as expected and remains relatively efficient.
```
Maybe:

```suggestion
spending still works as expected and remains relatively efficient and private.
```

Using a OP_CHECKSIG in a scriptpath is still more private than using OP_CHECKMULTISIG, since it doesn't reveal how many signers there are.

dongcarl commented on _posts/en/newsletters/2021-08-04-newsletter.md:
```diff
@@ -0,0 +1,83 @@
+  implementations we cover will have support for keysend.
+
+  The author has also submitted [corresponding documentation][BOLTs #892] (yet
+  unmerged) on keysend payments as a bLIP (Bitcoin Lightning Improvement
+  Proposals), a new way to document features and best practices which do not
+  belong as part of the spec.
```
Should we make it clear that bLIPs are still under discussion and not a sure thing yet?

harding replied:
That'd probably be good.  I think it might be sufficient to link to https://bitcoinops.org/en/newsletters/2021/07/07/#blips

Also, I'd prefer we capitalize it as "BLIP" (there's a comment with explanation in the source code of the above newsletter; I'll add it to the style guide if BLIPs do become a thing).

harding commented on _posts/en/newsletters/2021-08-04-newsletter.md:
```diff
@@ -63,7 +63,15 @@ guidelines for the addition of new tracepoints.
 
-- [Rust-Lightning #967][] valentinewallace/2021-06-keysend FIXME:dongcarl
+- [Rust-Lightning #967][] adds support for making [keysend-style spontaneous
+  payments][topic spontaneous payments#add-data-to-the-routing-packet] via the
```
Unfortunately markdown doesn't let you append a hash to a link descriptor like that.  You'll have to add a link definition pointing to `/en/topics/spontaneous-payments/#add-data-to-the-routing-packet` to the bottom of the page.

harding reviewed (commented):
Quick comments for @dongcarl on RL967.

harding commented:
Made edits for @jnewbery feedback (thanks!) and reviewed contributions by @jnewbery, @glozow, and @dongcarl --- all LGTM, thanks!  (One optional note left for Carl.)

dongcarl commented:
Changed:
- s/bLIP/BLIP/
- Linked to blip entry in 156
- s/a new way/a proposed way/

0xB10C commented:
Had a quick look at https://github.com/bitcoinops/bitcoinops.github.io/pull/617/commits/02e27c32df8538c7a93496e3d8c65ecea3fc86f9 (only this commit). LGTM. Thanks @glozow!

jnewbery commented on _posts/en/newsletters/2021-08-04-newsletter.md:
```diff
@@ -0,0 +1,85 @@
+{% include linkers/issues.md issues="22006,19866,1893,967,892" %}
+[C-Lightning 0.10.1]: https://github.com/ElementsProject/lightning/releases/tag/v0.10.1rc1
+[bitcoin core 22.0]: https://bitcoincore.org/bin/bitcoin-core-22.0/
+[tracing doc]: https://github.com/bitcoin/bitcoin/blob/master/doc/tracing.md
+[contrib tracing doc]: https://github.com/bitcoin/bitcoin/blob/master/contrib/tracing
```
I have a minor preference to linking to a specific commit (https://github.com/bitcoin/bitcoin/blob/84ace9aef116a05e034730f2bb2f109d1d77aac7/doc/tracing.md), which should never change, rather than master, which could change or disappear.

bitschmidty commented on _includes/specials/taproot/en/06-multisignatures.md:
```diff
@@ -0,0 +1,147 @@
+
+1. The wallet for each participant generates a [BIP32][] xpub that is shared
+   with all the other participants (the same as is commonly done now for
+   multisigs)
```
s/multisigs)/multisigs).

bitschmidty commented on _includes/specials/taproot/en/06-multisignatures.md:
```diff
@@ -0,0 +1,147 @@
+primary keys can use a multisignature for the taproot keypath spend (e.g.
+between Alice and Bob) and the additional outcomes (Alice and Carol, or
+Bob and Carol) can use multisignatures with the `OP_CHECKSIG` opcode in
+separate branches of a [tapscript][topic tapscript].
```
maybe something like s/separate branches of a tapscript./separate branches of a tree of tapscripts.

edit: I think I see what you mean. Using a single tapscript with OP_CHECKSIGs instead of 2 tapscript leaves in the taproot tree. If so maybe the word branches through me off.

harding replied:
Agree with suggestion, will fix.  FYI, I was thinking of two tapscript leaves.  The whole thing is:

       A+B
       /  \
    A+C   B+C

bitschmidty commented on _posts/en/newsletters/2021-08-04-newsletter.md:
```diff
@@ -0,0 +1,89 @@
+  The author has also submitted [corresponding documentation][BOLTs #892] (yet
+  unmerged) on keysend payments as a [BLIP][news156 blips] (Bitcoin Lightning
+  Improvement Proposals), a proposed way to document features and best practices
+  which do not belong as part of the spec.
```
s/the spec./the BOLT specs.

bitschmidty reviewed (commented):
Looks good, added a few comments. Apologies for the late review.

harding commented on _includes/specials/taproot/en/06-multisignatures.md:
```diff
@@ -0,0 +1,147 @@
+![Plot showing the savings for multisignatures compared to multisig](/img/posts/2021-07-multisignature-savings.png)
+
+The second major benefit is improved privacy.  Each use of multisigs is
+distinctively recorded to the block chain where surveilors
```
It's been used a few times in mailing list discussions, but some searching indicates a word that's in the dictionary (though I've never heard it used and Firefox's autocorrect doesn't like it) is surveillant.  https://duckduckgo.com/?t=ffab&q=surveillant&ia=definition I'll switch to that.

Xekyo reviewed (commented):
That's some fine content!

harding commented:
Edited for feedback from @bitschmidty and @Xekyo, plus additional feedback from @jnewbery (thanks all!).  Added topic links and updated the C-Lightning RC.

@bitschmidty Bitcoin Core hasn't yet released binaries for RC2.  Before merge tomorrow, could you check https://bitcoincore.org/bin/ to see if there's a `22.*` directory with RCs in it?  If so, confirm the link in the newsletter is correct.  If not, delete that bullet from the newsletter (I have it saved on a local branch and will bring it back next week, so no work will be wasted).

Thanks everyone!

0xB10C commented on _posts/en/newsletters/2021-08-04-newsletter.md:
```diff
@@ -82,6 +82,6 @@ guidelines for the addition of new tracepoints.
 [bitcoin core 22.0]: https://bitcoincore.org/bin/bitcoin-core-22.0/ {% comment %}<!-- FIXME: confirm when uploaded -->{% endcomment %}
 [tracing doc]: https://github.com/bitcoin/bitcoin/blob/master/doc/tracing.md
-[contrib tracing doc]: https://github.com/bitcoin/bitcoin/blob/master/contrib/tracing
+[contrib tracing doc]: https://github.com/bitcoin/bitcoin/blob/84ace9aef116a05e034730f2bb2f109d1d77aac7/doc/tracing.md
```
in 0ce1b67c0955bc3a7d847ff59f488d78b1bf42c1:

Both links ([tracing doc] and [contrib tracing doc]) now link to the same document. I think this should be

```markdown
[tracing doc]: https://github.com/bitcoin/bitcoin/blob/8f37f5c2a562c38c83fc40234ade9c301fc4e685/doc/tracing.md
[contrib tracing doc]: https://github.com/bitcoin/bitcoin/tree/8f37f5c2a562c38c83fc40234ade9c301fc4e685/contrib/tracing
```

harding replied:
Fixed, thanks @0xB10C!

glozow replied:
Thanks :)

jnewbery commented on _includes/specials/taproot/en/06-multisignatures.md:
```diff
@@ -0,0 +1,147 @@
+primary keys can use a multisignature for the taproot keypath spend (e.g.
+between Alice and Bob) and the additional outcomes (Alice and Carol, or
+Bob and Carol) can use multisignatures with the `OP_CHECKSIG` opcode in
+separate branches of a [tapscript][topic tapscript].
```
'leaves' seems slightly better to me, but I think both are fine.

jnewbery commented on _posts/en/newsletters/2021-08-04-newsletter.md:
```diff
@@ -0,0 +1,89 @@
+  associated wallet and other software. Major changes in this new
+  version include support for [I2P][topic anonymity networks] connections,
+  removal of support for [version 2 Tor][topic anonymity networks] connections,
+  support for [bech32m address encoding][topic bech32], and enhanced support for
```
Good spot. This wording is also in the 22.0 release notes (https://github.com/bitcoin-core/bitcoin-devwiki/wiki/22.0-Release-Notes-draft#updated-rpcs). I guess because it was backported from master to the 0.21.1 branch.

jnewbery commented on _includes/specials/taproot/en/06-multisignatures.md:
```diff
@@ -0,0 +1,147 @@
+   for the next transaction.
+
+3. Any of the wallets can then generate an aggregated public key by
+   combining its pubkey at a certain BIP32 depth with pubkeys at the
```
@Xekyo do you suggest any change to the wording here? I think this is probably fine as it is, and talking about descriptors here probably distracts from the more important concept.

jnewbery commented:
All LGTM. Thanks everyone!

bitschmidty commented:
Removing bitcoin core 0.22 bullet as the binaries have yet to drop

bitschmidty commented:
Squashed and merged. Good work team! 
@harding @glozow @dongcarl with the 🥇 
@jnewbery @0xB10C @Xekyo as the 🧑‍⚖️
